core/src/test/java/org/apache/iceberg/rest/TestHTTPClient.java

```diff
@@ -121,6 +128,92 @@ public void testHeadFailure() throws JsonProcessingException {
     testHttpMethodOnFailure(HttpMethod.HEAD);
   }
 
+  /** Tests that requests go via the proxy server in case the client is set up with one */
```

Comment:
Nit: While I personally like the idea of commenting tests typically in the project we don't add them and try to just write test names which are self evident in what they are testing. I think in this case just naming the test `testHttpClientProxyServer` is sufficient (no additional context needed I think for the happy case)